### PR TITLE
Add EdXOAuth2.auth_complete_signal on auth_complete()

### PR DESCRIPTION
**Pull request by pwnage101**
Also, version bumped to 2.0.2.

This is part of the DOP->DOT migration.

---

This is required for hooking into the DOT SSO workflow in edx-analytics-dashboard (insights).

### CODE REVIEW

File: auth_backends/backends.py [hunk 1]
```diff
@@ -290,6 +292,16 @@ def auth_complete_params(self, state=None):
         params['token_type'] = 'jwt'
         return params
 
+    def auth_complete(self, *args, **kwargs):
```
Comment (nasthagiri) [inline auth_backends/backends.py]: Can you explain why this is needed?

Comment (pwnage101) [reply auth_backends/backends.py]: Explain it in the code, or as a github comment?  It is needed during Insights auth to initiate the fetching of the "allowed" courses from the courses list API.

https://github.com/edx/edx-analytics-dashboard/pull/875/files#diff-d348b2b93c2b58adcb832a4e73324fc0R221

Comment (nasthagiri) [reply auth_backends/backends.py]: Ok. I understand now.

File: auth_backends/backends.py [hunk 2]
```diff
@@ -262,6 +262,8 @@ class EdXOAuth2(EdXBackendMixin, BaseOAuth2):
         'user_id': 'user_id',
     })
 
+    auth_complete_signal = Signal(providing_args=['user'])
```
Comment (nasthagiri) [inline auth_backends/backends.py]: Can you add a comment here to explain when this signal is fired?   For example, after the user has successfully logged in via OAuth Social Auth.

File: auth_backends/backends.py [hunk 3]
```diff
@@ -290,6 +292,16 @@ def auth_complete_params(self, state=None):
         params['token_type'] = 'jwt'
         return params
 
+    def auth_complete(self, *args, **kwargs):
+        """
+        Override this method to emit the `EdXOAuth2.auth_complete_signal` signal.
```
Comment (doctoryes) [inline auth_backends/backends.py]: Doesn't this method emit the mentioned signal below? One doesn't need to override the method to emit the signal - it already gets sent, correct?

Comment (pwnage101) [reply auth_backends/backends.py]: The wording could be clearer, I'll update it to say "We override this method to...".

File: auth_backends/backends.py [hunk 4]
```diff
@@ -295,6 +298,16 @@ def auth_complete_params(self, state=None):
         params['token_type'] = 'jwt'
         return params
 
+    def auth_complete(self, *args, **kwargs):
+        """
+        We override this method to emit the `EdXOAuth2.auth_complete_signal` signal.
```
Comment (doctoryes) [inline auth_backends/backends.py]: Nit:
`This method is overwritten to emit....`

**Pull request merged**